src/js/playlist.js

```diff
@@ -6,12 +6,11 @@ class Playlist {
 		const currentId = opts.player.videoData ? opts.player.videoData.id : opts.player.opts.id;
 		this.currentIndex = currentId ? opts.queue.indexOf(currentId.toString()) : -1;
 
-		this.opts.player.containerEl.addEventListener('ended', this.next.bind(this), true);
-
 		this.cache = {};
 
-		if (this.opts.autoplay && this.currentIndex === -1) {
-			this.next();
+		if (this.opts.autoplay) {
+			this.opts.player.containerEl.addEventListener('ended', this.next.bind(this), true);
+			this.currentIndex === -1 && this.next();
```

Comment:
I think this might confuse people who are not aware of the short-circuiting/lazy-evaluation with `&&`.

I think it might be easier to read as:
```
if (this.currentIndex === -1) {
   this.next()
}
```